### DISCUSSION TITLE
After using the module orion_node, how to get the Machine Type with the correct value?

### DISCUSSION THREAD

**Discussion by rnozaki** [jeisenbath/ansible-collection-solarwinds-orion#35] [Q&A] [answered]
I have used the orion_node to add a Windows VM to SolarWinds with Ansible, but differently from when we do it manually, the Machine Type does not get assigned.
I checked that the SNMP service is running and the community was passed when adding the node, and we assign the ID of a UnDP=2.
What should I need to do additionally to have the node showing up as a Windows box in SolarWinds, as it happens when we do it from the SolarWinds manual interface?

**Comment by jeisenbath** [comment]
In issue #20 we discuss adding both a poll now and returning a field to orion_node_info to confirm that a node is successfully polling with SNMP. You can install the updated modules by installing that devel branch right now, or I should be merging and updating main to 2.1.0 this week (main README has instructions on installing from a specific branch). I would start there to validate your node is polling after you create it, especially if you are not adding interfaces (orion_node_interface requires SNMP to be working to do a discovery). I think I'll add this type of task to the example playbook in 2.1.0, in case people aren't familiar with using the "until" task parameter.

**Comment by rnozaki** [comment]
Turns out that my issue was much simpler... a typo in the snmp credential that I added to the AAP credential.
But yes... being able to for a first poll and make sure that the snmp queries are working would be great.
On the SolarWinds GUI they show the status of the node as being Up, but now I realize that this was only through ICMP, not SNMP.

**Reply by jeisenbath** [reply, marked as answer]
I finally got around to completing validation and released version 2.1.0, the example playbook orion_add_node.yml now has an task in it that should help validating SNMP is working.
